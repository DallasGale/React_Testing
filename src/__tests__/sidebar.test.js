import Sidebar from '../containers/sidebar.container';
import Links from '../components/links.component';
import SidebarComponent from '../components/sidebar.component';
import Toggle from '../components/toggle.component';
// - - - - - - - - - - - - - - - - - - - - - 
// using Enzyme to step through the DOM tree
// - - - - - - - - - - - - - - - - - - - - - 

// Shallow 

it("renders correctly", () => {
  const wrapper = shallow(
    <Sidebar click={ () => {} } open={ false } />
  );
  expect(wrapper).toMatchSnapshot();
});


// Rendered ie. HTML child

it("renders correctly again", () => {
  const wrapper = render(
    <Sidebar click={ () => {} } open={ false } />
  );
  expect(wrapper).toMatchSnapshot();
});





it('responds to toggle click', () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <Toggle click={ spy } label='>' />
  );
  
  wrapper
    .simulate('click');
    console.log(spy.calledOnce);
  expect(spy.calledOnce).toBe(true);
});




describe('<SidebarComponent />', () => {
  
  it('renders correctily', () => {
    const wrapper = shallow(<SidebarComponent />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders 3 <Links /> components', () => {
    const wrapper = shallow(<SidebarComponent />);
    expect(wrapper.find(Links).length).toBe(3);
  });
});




 