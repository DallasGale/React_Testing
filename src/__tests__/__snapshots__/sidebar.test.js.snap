// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarComponent /> renders correctily 1`] = `
<aside
  className="css-1pp3ebs"
>
  <nav>
    Navigation
    <ul
      className="css-zkg6nm"
    >
      <Links
        key="0"
        name="HOME"
      />
      <Links
        key="1"
        name="ABOUT"
      />
      <Links
        key="2"
        name="CONTACT"
      />
    </ul>
  </nav>
</aside>
`;

exports[`renders correctly 1`] = `
<div>
  <ToggleSidebarContainer
    click={[Function]}
    open={true}
  />
  <Sidebar
    styles={
      Object {
        "left": "0",
      }
    }
  />
</div>
`;

exports[`renders correctly again 1`] = `
<div>
  <button
    class="css-1pa205l"
  >
    &lt;
  </button>
  <aside
    class="css-1pp3ebs"
    style="left:0"
  >
    <nav>
      Navigation
      <ul
        class="css-zkg6nm"
      >
        <li
          class="css-8wiwd4"
        >
          <a
            href="home/"
          >
            HOME
          </a>
        </li>
        <li
          class="css-8wiwd4"
        >
          <a
            href="about/"
          >
            ABOUT
          </a>
        </li>
        <li
          class="css-8wiwd4"
        >
          <a
            href="contact/"
          >
            CONTACT
          </a>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;
